library/blob.c

```diff
@@ -1057,6 +1053,83 @@ static int eblob_blob_iterator(struct eblob_iterate_priv *iter_priv)
 	return err;
 }
 
+
+/**
+ * eblob_disk_control_cmp_by_position() - compares dc's by position in a blob file
+ */
+static int eblob_disk_control_cmp_by_position(const void *lhs, const void *rhs) {
+	const struct eblob_disk_control *left = (const struct eblob_disk_control *)lhs;
+	const struct eblob_disk_control *right = (const struct eblob_disk_control *)rhs;
+	if (left->position == right->position)
+		return 0;
+
+	return left->position < right->position ? -1 : 1;
+}
+
+/**
+ * eblob_blob_iterate_by_pos() - read whole index, sort it by position and run iteration over it.
+ */
+static int eblob_blob_iterate_by_pos(struct eblob_iterate_priv *iter_priv)
+{
+	struct eblob_iterate_control *ctl = iter_priv->ctl;
+	struct eblob_base_ctl *bctl = ctl->base;
+
+	struct eblob_disk_control *dc = NULL;
+	struct eblob_iterate_local loc;
+	int err = 0;
+
+	static const int hdr = sizeof(struct eblob_disk_control);
+
+	if (iter_priv == NULL) {
+		err = -EINVAL;
+		goto err_out_check;
+	}
+
+	memset(&loc, 0, sizeof(loc));
+
+	loc.iter_priv = iter_priv;
+	loc.index_offset = 0;
+	loc.pos = 0;
+	loc.num = bctl->index_ctl.size / hdr;
+
+	dc = malloc(bctl->index_ctl.size);
+	if (dc == NULL) {
+		err = -ENOMEM;
+		goto err_out_check;
+	}
+
+	pthread_mutex_lock(&bctl->lock);
+	err = __eblob_read_ll(bctl->index_ctl.fd, dc, bctl->index_ctl.size, 0);
+	pthread_mutex_unlock(&bctl->lock);
+	if (err) {
+		goto err_out_check;
+	}
+
+	qsort(dc, loc.num, hdr, eblob_disk_control_cmp_by_position);
+	loc.dc = dc;
+
+	eblob_bctl_hold(bctl);
+	err = eblob_check_disk(&loc);
+	eblob_bctl_release(bctl);
+
+err_out_check:
+	free(dc);
+	eblob_log(ctl->log, err < 0 ? EBLOB_LOG_ERROR : EBLOB_LOG_INFO, "blob-0.%d: iterated: data_fd: %d, index_fd: %d, "
+									"data_size: %llu, index_offset: %llu, err: %d\n",
+		  bctl->index, bctl->data_ctl.fd, bctl->index_ctl.fd, ctl->data_size, ctl->index_offset, err);
+
+	/*
```

Comment:
As we don;t use threads anymore this comment looks outdated

Reply:
removed